src/parser.mjs

```diff
@@ -596,11 +597,17 @@ export class ExifParser extends Reader {
 
 
 	parseXmpSegment() {
-		// Cancel if the file doesn't contain the segment or if it's damaged.
-		if (!this.ensureSegmentPosition('xmp', findXmp)) return
-
-		// Read XMP segment as string. We're not parsing the XML.
-		this.xmp = toString(this.buffer, this.xmpOffset, this.xmpOffset + this.xmpEnd)
+		if (this.ensureSegmentPosition('xmp', findXmp)) {
```

Comment:
This is a more general way of extracting XMP from ApplicationNotes. None of the current test images have both XMP and ApplicationNotes, so merging should not be needed.

Reply:
Dont we need to check this.options.xmp also here?

Reply:
If not the why is it in the second if

Reply:
It shouldn't be in the second if, good catch. The parse function only calls this function when `options.xmp` is true.

Reply:
Fixed it, but still have to push